corehq/apps/linked_domain/applications.py

```diff
@@ -44,8 +48,17 @@ def get_master_app_by_version(domain_link, upstream_app_id, upstream_version):
         return wrap_app(app)
 
 
-def get_downstream_app_id(downstream_domain, upstream_app_id):
-    downstream_ids = _get_downstream_app_id_map(downstream_domain)[upstream_app_id]
+def get_downstream_app_id(downstream_domain, upstream_app_id, use_upstream_app_id=False):
```

Comment:
Would it be arduous to make `use_upstream_app_id` default to `True` since that's the desired usage, and update other existing calls to this function?

Reply:
Nope and it is a good idea.